File: algorithms/11_golang/techniques/sliding_window.go
package main

import "fmt"

func maxSumFixedWindow(nums []int, k int) int {
	if k > len(nums) {
		return 0
	}
	sum := 0
	for i := 0; i < k; i++ {
		sum += nums[i]
	}
	best := sum
	for i := k; i < len(nums); i++ {
		sum += nums[i] - nums[i-k]
		if sum > best {
			best = sum
		}
	}
	return best
}

func main() {
	nums := []int{2, 1, 5, 1, 3, 2}
	if maxSumFixedWindow(nums, 3) != 9 {
		panic("sliding window: fixed-size max sum failed")
	}
	fmt.Println("[Go SlidingWindow] Fixed-size max sum verified: 9")
}

File: algorithms/02_c/techniques/sliding_window.c
#include <stdio.h>

int max_subarray_sum(const int *a, int n, int k) {
    int sum = 0;
    for (int i = 0; i < k; ++i) sum += a[i];
    int best = sum;
    for (int i = k; i < n; ++i) {
        sum += a[i] - a[i - k];
        if (sum > best) best = sum;
    }
    return best;
}

int main(void) {
    int a[] = {1, 4, 2, 10, 23, 3, 1, 0, 20};
    int n = sizeof(a) / sizeof(a[0]);

    if (max_subarray_sum(a, n, 4) != 39) {
        printf("[C SlidingWindow] FAILED: max fixed-window sum\n");
        return 1;
    }
    printf("[C SlidingWindow] Fixed-size window max sum verified\n");
    return 0;
}
